src/iota/transfers.c

```diff
@@ -45,171 +25,299 @@ static char *int64_to_chars(int64_t value, char *chars, unsigned int num_trytes)
+    c = int64_to_chars(tx->currentIndex, c, 9);
+    c = int64_to_chars(tx->lastIndex, c, 9);
+    c = char_copy(c, bundle_hash, 81);
+    c = char_copy(c, ZERO_HASH, 81);
+    c = char_copy(c, ZERO_HASH, 81);
+    c = char_copy(c, tx->tag, 27);
+    c = int64_to_chars((int64_t) 0, c, 9);
+    c = int64_to_chars((int64_t) 0, c, 9);
+    c = int64_to_chars((int64_t) 0, c, 9);
+    char_copy(c, tx->nonce, 27);
+}
+
+iota_wallet_tx_object_t tx_object_buffer = {};
 
-    for (unsigned int i = 0; i < num_txs; i++) {
-        memcpy(txs[i].bundle, bundle, 81);
+void clear_tx_object_buffer(void){
+    char_copy(tx_object_buffer.address, ZERO_HASH, 81);
+    char_copy(tx_object_buffer.branchTransaction, ZERO_HASH, 81);
+    char_copy(tx_object_buffer.trunkTransaction, ZERO_HASH, 81);
+    char_copy(tx_object_buffer.bundle, ZERO_HASH, 81);
+    char_copy(tx_object_buffer.nonce, ZERO_TAG, 27);
+    char_copy(tx_object_buffer.tag, ZERO_TAG, 27);
+    char_copy(tx_object_buffer.obsoleteTag, ZERO_TAG, 27);
+
+    tx_object_buffer.timestamp = 0;
+    tx_object_buffer.value = 0;
+
+
+    for(unsigned int i = 0; i < 2187; i++){
+        tx_object_buffer.signatureMessageFragment[i] = '9';
     }
 }
 
-void prepare_transfers(char *seed, uint8_t security, TX_OUTPUT *outputs,
-                       int num_outputs, TX_INPUT *inputs, int num_inputs,
-                       char transaction_chars[][2673])
-{
-    // TODO use a proper timestamp
-    const uint32_t timestamp = 0;
+void clear_tx_signature_bytes_buffer(void){
+    for(unsigned int i = 0; i < 2187; i++){
+        tx_object_buffer.signatureMessageFragment[i] = '9';
+    }
+}
+
+
+iota_wallet_tx_receiver_t tx_receiver_func;
+void iota_wallet_set_tx_receiver_func(iota_wallet_tx_receiver_t func){
+    tx_receiver_func = func;
+}
+
+iota_wallet_bundle_hash_receiver_ptr_t bundle_hash_receiver_func;
+void iota_wallet_set_bundle_hash_receiver_func(iota_wallet_bundle_hash_receiver_ptr_t func){
+    bundle_hash_receiver_func = func;
+}
+
+unsigned char seed_bytes_buffer[48];
+void clear_seed_bytes_buffer(void){
+    for(int i = 0; i < 48; i++){
+        seed_bytes_buffer[i] = 0;
+    }
+}
+
+tryte_t normalized_bundle_hash_buffer[81];
+void clear_normalized_bundle_hash_buffer(void){
+    for(int i = 0; i < 81; i++){
+        normalized_bundle_hash_buffer[i] = '9';
+    }
+}
+
+
+unsigned char signature_bytes_buffer[27 * 48];
+void clear_signature_bytes_buffer(void){
+    for(int i = 0; i < 27 * 48; i++){
+        signature_bytes_buffer[i] = 0;
+    }
+}
+
+// create a secure bundle
+BUNDLE_CTX bundle_ctx_buffer;
+void clear_bundle_ctx_buffer(void){
+    bundle_ctx_buffer.last_index = 0;
+    bundle_ctx_buffer.current_index = 0;
+
+    for(int i = 0; i < 768; i++){
+        if(i < 8){
+            bundle_ctx_buffer.values[i] = 0;
+            bundle_ctx_buffer.indices[i] = 0;
+        }
+        if(i < 48){
+            bundle_ctx_buffer.hash[i] = 0;
+        }
+
+        bundle_ctx_buffer.bytes[i] = 0;
+    }
```

Comment:
memset would look much nicer. At least, this should be done without "magic values"